### PR TITLE
Move cs_main locking annotations from .cpp to .h

### PR DESCRIPTION
**Pull request by practicalswift**
Move `cs_main` locking annotations from `.cpp` to `.h`.

### CODE REVIEW

**Comment by practicalswift** [comment]
As suggested by @MarcoFalke :-)

File: src/txmempool.h [hunk 1]
```diff
@@ -365,6 +365,8 @@ class SaltedTxidHasher
     }
 };
 
+extern CCriticalSection cs_main;
```
Comment (MarcoFalke) [inline src/txmempool.h]: Could move this all the way up (right after the includes?)

**Comment by practicalswift** [comment]
@MarcoFalke Feedback addressed. Please re-review :-)

**Comment by laanwj** [comment]
does this make any difference to the functionality? or is this just better from a documentation point of view?

**Comment by MarcoFalke** [comment]
The locking annotations are only evaluated at compile time, so when they are not in the header file, they are ignored outside of that translation unit.

**Pull request merged**